spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultPubSubAcknowledger.java

```diff
@@ -28,11 +28,11 @@
  *
  * @author João André Martins
  */
-class DefaultPubSubAcknowledger implements PubSubAcknowledger {
+public class DefaultPubSubAcknowledger implements PubSubAcknowledger {
```

Comment:
Why must it become as a `public`?

Seems for me the idea around a `PubSubAcknowledger` abstraction that end-user may decide to implement his/her own customization in his/her own `SubscriberFactory`. So, we have only single place to control a subscriber and `acknowledger`. This way the `PubSubSubscriberTemplate` is free from any implementation details and we have a good distribution of responsibility and dependency injection power.

Yes, template could factor an instance from the `createSubscriberStub()`, but if we don't do that but just delegate to the `SubscriberFactory`, everything is good and no one in a harm.

The `async` option is really a `DeafultSubscriberFactory` responsibility: I don't see any issues on the matter...

So, -1 from me for this change